subsys/event_manager/event_manager_priv.h

```diff
@@ -49,31 +49,22 @@ extern "C" {
 
 /* Convenience macros generating section start and stop markers. */
 
-#define _EVENT_SUBSCRIBERS_START(ename, prio)	_CONCAT(__start_, _EVENT_SUBSCRIBERS_SECTION_PREFIX(ename, prio))
+#define _EVENT_SUBSCRIBERS_START(ename)	_CONCAT(__start_, ename)
 
-#define _EVENT_SUBSCRIBERS_STOP(ename, prio)	_CONCAT(__stop_,  _EVENT_SUBSCRIBERS_SECTION_PREFIX(ename, prio))
+#define _EVENT_SUBSCRIBERS_STOP(ename)	_CONCAT(__stop_, ename)
 
 
-#define _EVENT_SUBSCRIBERS_DECLARE(ename)										\
-	extern const struct event_subscriber _EVENT_SUBSCRIBERS_START(ename, _SUBS_PRIO_ID(_SUBS_PRIO_FIRST))[];	\
-	extern const struct event_subscriber _EVENT_SUBSCRIBERS_STOP(ename,  _SUBS_PRIO_ID(_SUBS_PRIO_FIRST))[];	\
-	extern const struct event_subscriber _EVENT_SUBSCRIBERS_START(ename, _SUBS_PRIO_ID(_SUBS_PRIO_NORMAL))[];	\
-	extern const struct event_subscriber _EVENT_SUBSCRIBERS_STOP(ename,  _SUBS_PRIO_ID(_SUBS_PRIO_NORMAL))[];	\
-	extern const struct event_subscriber _EVENT_SUBSCRIBERS_START(ename, _SUBS_PRIO_ID(_SUBS_PRIO_FINAL))[];	\
-	extern const struct event_subscriber _EVENT_SUBSCRIBERS_STOP(ename,  _SUBS_PRIO_ID(_SUBS_PRIO_FINAL))[];
+#define _EVENT_SUBSCRIBERS_DECLARE(ename)						\
+	extern const struct event_subscriber _EVENT_SUBSCRIBERS_START(ename)[];		\
+	extern const struct event_subscriber _EVENT_SUBSCRIBERS_STOP(ename)[];		\
 
 
-/* Macro defining empty subscribers on each priority level.
- * Each event type keeps an array of subscribers for every priority level.
- * It can happen that for a given priority no subscriber will be registered.
- * This macro declares zero-length subscriber that will cause required section
- * to be generated by the linker. If no subscriber is registered at this
- * level section will remain empty.
+/* Macro defining two empty subscribers for each event type.
+ * Each event type keeps an array of subscribers starting with empty .
  */
-#define _EVENT_SUBSCRIBERS_DEFINE(ename)					\
-	_EVENT_SUBSCRIBERS_EMPTY(ename, _SUBS_PRIO_ID(_SUBS_PRIO_FIRST))	\
-	_EVENT_SUBSCRIBERS_EMPTY(ename, _SUBS_PRIO_ID(_SUBS_PRIO_NORMAL))	\
-	_EVENT_SUBSCRIBERS_EMPTY(ename, _SUBS_PRIO_ID(_SUBS_PRIO_FINAL))
+#define _EVENT_SUBSCRIBERS_DEFINE(ename)	\
+	_EVENT_SUBSCRIBERS_EMPTY(ename, _a)	\
```

Comment:
I think we need to hide `_a` `_z` and `_prio` under some macro. I also think that "final" object should have it's own marker (that would be `_y` just before the stop marker).

Reply:
Why should final object have its own marker, isn't it just like any other object that would be placed at the end? Or is it about making some assert which will find out if we only have one final subscriber?

Reply:
if we do it with a marker we ensure that it is at the end without playing with level values

order on the section would be:
section_prefix_a - start tag
section_prefix_b - first element (only one)
section_prefix_p00 - normal elements ordered by priorities
(...)
section_prefix_p99 - normal elements ordered by priorities
section_prefix_y - final element (only one)
section_prefix_z - end tag

assert that we have one cannot be done this way - we create a zero sized object to make linker fail in case somebody defines more then one. We do it in event_manager.h.